platforms/core-runtime/native/src/main/java/org/gradle/internal/nativeintegration/services/NativeServices.java

```diff
@@ -322,28 +324,27 @@ private static String getNativeDirOverride() {
         return System.getProperty(NATIVE_DIR_OVERRIDE, System.getenv(NATIVE_DIR_OVERRIDE));
     }
 
-    public static synchronized NativeServices getInstance() {
+    public static synchronized ServiceRegistry getInstance() {
         if (!INSTANCE.initialized) {
             // If this occurs while running gradle or running integration tests, it is indicative of a problem.
             // If this occurs while running unit tests, then either use the NativeServicesTestFixture or the '@UsesNativeServices' annotation.
             throw new IllegalStateException("Cannot get an instance of NativeServices without first calling initialize().");
         }
-        return INSTANCE;
+        return INSTANCE.services;
```

Comment:
Why not have `INSTANCE` be the `ServiceRegistry`?

Reply:
This is what I initially tried, but because of the [deferred initialization](https://github.com/gradle/gradle/blob/6a61e6d812b2cefb7fb529f62ab4c867ca80c1d7/platforms/core-runtime/native/src/main/java/org/gradle/internal/nativeintegration/services/NativeServices.java#L172-L174) logic it does not work nicely.

![image](https://github.com/gradle/gradle/assets/2759152/8e761a30-8f9d-4e7c-b9a4-d8255445cf45)